pkg/neotest/coverage.go

```diff
@@ -0,0 +1,153 @@
+package neotest
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/nspcc-dev/neo-go/pkg/compiler"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+)
+
+var rawCoverage = make(map[scriptHash]*scriptRawCoverage)
+
+var enabled = false
+var coverProfile = ""
+
+type scriptHash = util.Uint160
+
+type scriptRawCoverage struct {
+	debugInfo      *compiler.DebugInfo
+	offsetsVisited []int
+}
+
+type coverBlock struct {
+	startLine uint // Line number for block start.
+	startCol  uint // Column number for block start.
+	endLine   uint // Line number for block end.
+	endCol    uint // Column number for block end.
+	stmts     uint // Number of statements included in this block.
+	counts    uint
+}
+
+type documentName = string
+
+func isCoverageEnabled() bool {
+	if enabled {
+		return true
+	}
+	const coverProfileFlag = "test.coverprofile"
+	flag.VisitAll(func(f *flag.Flag) {
+		if f.Name == coverProfileFlag && f.Value != nil {
+			enabled = true
+			coverProfile = f.Value.String()
+		}
+	})
+	if enabled {
+		// this is needed so go cover tool doesn't overwrite
+		// the file with our coverage when all tests are done
+		flag.Set(coverProfileFlag, "")
```

Comment:
This is quite a hack and it could violate some `go test` internal invariants, in theory.
Could you describe other solutions you tried?

Reply:
In theory yes, but in practice as of Go 1.20 `coverProfile` is used before all tests are run (see image) and after all tests were run (to write coverage) 
![image](https://github.com/nspcc-dev/neo-go/assets/40753025/eb7e0e3c-5ebc-457e-a2f2-58d1eaa53c85)

The other solution is writing `go test` wrapper that rewrites coverage report file after it finishes, but this would require more effort from user.